### ISSUE TITLE
onSubmit check

### ISSUE THREAD

**Issue by romseguy** [wsmd/react-use-form-state#34]
Consider an email/password form. I type in the email, TAB, type in the password, ENTER.
We get this state from onSubmit callback:
![submit](https://user-images.githubusercontent.com/109238/54189903-0d086700-44b3-11e9-907f-032f3db21bc2.png)
Which is logical since pressing ENTER doesn't trigger a blur event. Is it a good idea to let users implement this themselves instead of just being able to write:

```javascript
  const hasClientErrors = !all(v => v === true, values(formState.validity))
  const onSubmit = () => {
    if (hasClientErrors) return
    onSubmit()
  }
```

EDIT: actually it's about using `onKeyPress` prop on controls so we can trigger validity check on ENTER. Maybe it would be nice to have it included with props builders  `<input {...email('email')} />`

**Comment by wsmd** [comment]
Hi @romseguy,

I'm curious what version of `react-use-form-state` are you currently using? In the latest version input validation happens by default `onChange` as well.

Here's a demo: https://codesandbox.io/s/24394n3v2j

**Comment by romseguy** [comment]
Ah yes, you've fixed it with 0.8.0, well done :smile:
